### PR TITLE
Remove Ubuntu 14 actions and rules

### PR DESCRIPTION
**Pull request by blag**

### CODE REVIEW

File: CHECKLIST.md [hunk 1]
```diff
@@ -38,15 +40,14 @@ Valid for minor releases. For patches see [RELEASE_HOWTO.md](https://github.com/
...
+- [ ] Add new version in [st2docs/docs/source/conf.py](https://github.com/StackStorm/st2docs/blob/master/docs/source/conf.py#L79) e.g. if `1.3.0` is release then `1.3` should be added to the list.
 - [ ] Set the version values in the KV store of the build box:
 ```
-st2 key set st2_stage_v1.2_UBUNTU14 ubuntu-staging202
 st2 key set st2_stable_version 1.2
 st2 key set st2_unstable_version 1.3dev
 st2 key set st2_master_build_number 1
 ```
-- [ ] Make sure packaging is completed for st2web, st2flow, Mistral, and st2 itself (x3: Ubuntu 14, RHEL6, RHEL7).
+- [ ] Make sure packaging is completed for st2web, st2flow, Mistral, and st2 itself (x4: RHEL6, RHEL7).
```
Comment (cognifloyd) [inline CHECKLIST.md]: Did you mean to include Ubuntu 16, 18 here as well?

File: actions/setup_e2e_tests.sh [hunk 2]
```diff
@@ -86,9 +84,7 @@ elif [[ -n "$DEBTEST" ]]; then
     fi
 
     # Install from GitHub
-    # Ubuntu 16.04 has both bats and jq packages, so we don't need to do this
-    # once we drop Ubuntu 14.04 support
-    git clone --branch add_per_test_timing_information --depth 1 https://github.com/Kami/bats-core.git
+    git clone https://github.com/bats-core/bats-core.git
```
Comment (cognifloyd) [inline actions/setup_e2e_tests.sh:87]: Is the bats 0.4.0 deb too old? The previous comment suggests that we won't need the repo once we move off 14.04.

**Review by cognifloyd — COMMENTED** [review]
a couple questions

**Comment by nmaludy** [comment]
Looking good so far!

File: actions/setup_e2e_tests.sh (hunk 2 above)
Comment (blag) [reply actions/setup_e2e_tests.sh:87]: To be honest, I'm not 100% sure. I figured we could try out that change in a future PR - I just wanted to make the quickest change to remove Ubuntu 14 support. I'll create an issue to track this separately.

File: CHECKLIST.md (hunk 1 above)
Comment (blag) [reply CHECKLIST.md]: Yep, thanks for catching that.

File: actions/setup_e2e_tests.sh (hunk 2 above)
Comment (blag) [reply actions/setup_e2e_tests.sh:87]: Tracked in https://github.com/StackStorm/st2cd/issues/434 - resolving.

**Pull request merged**